services/logger/messenger/src/lib/MessengerFactory.js

```diff
@@ -0,0 +1,95 @@
+import AWS from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
+import connectionClass from 'http-aws-es';
+import elasticsearch from 'elasticsearch';
+import Logger from '@eubfr/logger-listener/src/lib/Logger';
+
+// Env vars
+const { META_ENDPOINT, META_INDEX, REGION, STAGE } = process.env;
+
+const shouldPersist = message => !!message.persist;
+const getAccountId = context => context.invokedFunctionArn.split(':')[4];
+const getClients = context => {
+  // AWS clients
+  const sns = new AWS.SNS();
+
+  const accountId = getAccountId(context);
+
+  this.loggerIndexSnsEndpointArn = `arn:aws:sns:${REGION}:${accountId}:${STAGE}-onLogEmitted`;
+  this.metaIndexSnsEndpointArn = `arn:aws:sns:${REGION}:${accountId}:${STAGE}-MetaStatusReported`;
+
+  this.logsLoggerclient = new Logger({
+    sns,
+    targetArn: this.loggerIndexSnsEndpointArn,
+    emitter: context.invokedFunctionArn,
+  });
+
+  this.metaIndexClient = elasticsearch.Client({
+    host: `https://${META_ENDPOINT}`,
+    apiVersion: '6.0',
+    connectionClass,
+    index: META_INDEX,
+  });
+
+  return {
+    logs: {
+      logger: this.logsLoggerclient,
+    },
+    meta: {
+      es: this.metaIndexClient,
+      sns,
+    },
+  };
+};
+
+const MessengerFactory = {
+  Create({ context }) {
+    MessengerFactory.context = context;
+    MessengerFactory.clients = getClients(context);
+
+    return Object.create(this.messenger);
+  },
+
+  messenger: {
+    send({ message, to }) {
```

Comment:
No `async`? The function doesn't return anything?

Reply:
The implementation `awaits` (resolves) the promises in the `send` without return. I'm not too deep into async/await to know why though